manager/src/main/java/com/redhat/service/bridge/manager/BridgeWorker.java

```diff
@@ -0,0 +1,208 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BridgeWorker.class);
+
+    private static final String CREATE_KAFKA_TOPIC = "bridge-create-topic";
+    private static final String DELETE_KAFKA_TOPIC = "bridge-delete-topic";
+
+    @Inject
+    EventBus eventBus;
+
+    @Inject
+    BridgeDAO bridgeDAO;
+
+    @Inject
+    PreparingWorkerDAO preparingWorkerDAO;
+
+    @Inject
+    RhoasService rhoasService;
+
+    @Inject
+    InternalKafkaConfigurationProvider internalKafkaConfigurationProvider;
+
+    @Inject
+    WorkerIdProvider workerIdProvider;
+
+    @Override
+    public void accept(Bridge bridge) {
+        LOGGER.info("Accept entity " + bridge.getId());
+        if (bridge.getStatus() != null) {
+            LOGGER.warn("The status of the entity is not null. It will be overwritten.");
+        }
+        bridge.setStatus(ManagedEntityStatus.ACCEPTED);
+        bridge.setDesiredStatus(ManagedEntityStatus.PREPARING);
+
+        // Persist and fire
+        transact(bridge);
+
+        PreparingWorker preparingWorker = new PreparingWorker();
+        preparingWorker.setEntityId(bridge.getId());
+        preparingWorker.setStatus(null); // where the resource is
+        preparingWorker.setDesiredStatus(BridgeWorkerStatus.TOPIC_CREATED.toString()); // where you want to go. i.e. step 1
+        preparingWorker.setWorkerId(workerIdProvider.getWorkerId()); // The workerId is set immediately
+        preparingWorker.setSubmittedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        preparingWorker.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        preparingWorker.setType("BRIDGE");
+        transact(preparingWorker);
+
+        routePreparingWorker(preparingWorker, bridge);
+    }
+
+    @Override
+    public void deprovision(Bridge bridge) {
+        LOGGER.info("Deprovision entity " + bridge.getId());
+
+        bridge.setStatus(ManagedEntityStatus.DEPROVISION_ACCEPTED);
+        bridge.setDesiredStatus(ManagedEntityStatus.DEPROVISION);
+
+        // Persist and fire
+        transact(bridge);
+
+        PreparingWorker preparingWorker = new PreparingWorker();
+        preparingWorker.setEntityId(bridge.getId());
+        preparingWorker.setStatus(null); // where the resource is
+        preparingWorker.setDesiredStatus(BridgeWorkerStatus.TOPIC_DELETED.toString()); // where you want to go
+        preparingWorker.setWorkerId(workerIdProvider.getWorkerId());
+        preparingWorker.setSubmittedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        preparingWorker.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        transact(preparingWorker);
+
+        routePreparingWorker(preparingWorker, bridge);
+    }
+
+    @Transactional
+    @ConsumeEvent(value = CREATE_KAFKA_TOPIC, blocking = true)
+    public void step1_createKafkaTopic(Bridge bridge) {
+        try {
+            rhoasService.createTopicAndGrantAccessFor(internalKafkaConfigurationProvider.getBridgeTopicName(bridge), RhoasTopicAccessType.CONSUMER_AND_PRODUCER);
```

Comment:
We need to ensure that `RhoasService` is idempotent, otherwise we will always fail if we have to retry. Imagine the following scenario:

1. First time we create the Topic, but fail to create the ACL (what we're seeing at the minute a lot)
2. Second time, we fail because the Topic already exists
3. Repeat

`RhoasService` needs to be able to create the ACL if the requested topic already exists.

Reply:
it is idempotent.